contracts/PBAB+Collabs/GenArt721CoreV2_PBAB_FLEX.sol

```diff
@@ -0,0 +1,899 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../interfaces/0.8.x/IRandomizer.sol";
+import "../interfaces/0.8.x/IGenArt721CoreV2_PBAB.sol";
+
+import "@openzeppelin-4.5/contracts/utils/Strings.sol";
+
+import "@openzeppelin-4.5/contracts/token/ERC721/ERC721.sol";
+
+pragma solidity 0.8.9;
+
+/**
+ * @title Powered by Art Blocks ERC-721 core contract.
+ * @author Art Blocks Inc.
+ */
+contract GenArt721CoreV2_PBAB_FLEX is ERC721, IGenArt721CoreV2_PBAB {
+    /// randomizer contract
+    IRandomizer public randomizerContract;
+
+    struct Project {
+        string name;
+        string artist;
+        string description;
+        string website;
+        string license;
+        string projectBaseURI;
+        uint256 invocations;
+        uint256 maxInvocations;
+        string scriptJSON;
+        mapping(uint256 => string) scripts;
+        uint256 scriptCount;
+        string ipfsHash;
+        bool active;
+        bool locked;
+        bool paused;
+        bool externalAssetDependenciesLocked;
+        uint256 externalAssetDependencyCount;
```

Comment:
follow-on nit: re: lessons learned here: https://github.com/ArtBlocks/artblocks-contracts/pull/248, recommend making this `uint24 externalAssetDependencyCount`, and casting to/from uint256 at points of external interfaces.

I don't think we need to fully optimize/pack this struct, but when we do that in a V4, I think this will certainly be a uint24, so starting with that will make things more consistent in the future.

Reply:
🙏 ➕

Reply:
updated to use uint24 internally, but the public convenience getter function `projectExternalAssetDependencyCount` casts and returns uint256 still